async/abortable.ts

```diff
@@ -4,10 +4,15 @@
 import { createAbortError } from "./_util.ts";
 
 /**
- * Make {@linkcode Promise} abortable with the given signal.
+ * Make a {@linkcode Promise} abortable with the given signal.
+ *
+ * @typeParam T The type of the provided and returned promise.
+ * @param p The promise to make abortable.
+ * @param signal The signal to abort the promise with.
+ * @returns A promise that can be aborted.
  *
  * @example
```

Comment:
Nit: It would've been nice to have a snippet title to avoid having it as "Example 1" in JSR. Ditto for other `@example` code snippets.

Reply:
i feel like if its a single example, it being `Example 1` is fine. most scenarios have only really one example that makes sense to have, and describing it will seem a bit too much. though definitively should be enforced for multiple examples

Reply:
We've now added example titles to the list of checks for the doc checker. Please feel free to review #4841.